### ISSUE TITLE
[Question] How is @EntityId supposed to work on IntBags?

### ISSUE THREAD

**Issue by DaanVanYperen** [junkdog/artemis-odb#574]
@Junkdog Another one after running the debugger ;)

**Observed** `@EntityId` with CHECK_SOURCE_AND_TARGETS cleans up ints and IntBags, on EntityLinkManager proces.
**Expected** `@EntityId` with CHECK_SOURCE_AND_TARGETS cleans up deleted entities, each time a system finishes processing.

Is this a regression or is it intended? What use is CHECK_SOURCE?

Tests for reproducing the above:
https://gist.github.com/DaanVanYperen/507888e83d9b1ff9d35810f671595360

**Comment by junkdog** [comment]
I'll take a closer look tomorrow, but I think it's intentional. IIrc - only clean-up/callback when entity has been purged by the EntityManager, otherwise entities may be evicted prematurely (ie; "some" time before EntiySubcriptions have fired)

**Comment by DaanVanYperen** [comment]
Purges are after each system invocation right? System1->purge->System2->etc.

Maybe I'm doing something wrong. As it is now, when I delete a referenced entity in System1, the @EntityId annotated reference continues to dangle throughout System2..SystemN until the EntityLinkManager is invoked and it is finally cleared.

**Comment by junkdog** [comment]
Ah, sorry, purge as the final step in process - ComponentManager and EntityManager purging/resetting

**Comment by junkdog** [comment]
Ouch. Yeah, sounds like a bug with impl.

**Comment by DaanVanYperen** [comment]
Maybe `@LinkPolicy` javadoc can be clarified a bit as well. Especially how each option interacts with `@EntityId`.

**Comment by junkdog** [comment]
Yeah, I need to brush up on the inner workings as well... (a testament for the need for good documentation!)

**Comment by DaanVanYperen** [comment]
Delaying this one release for when we have more breaking changes (gdx 1.10 etc).
